megasync: 4.9.1.0 -> 5.6.1.0

hackerncoder opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Updated megasync to 5.6.1.0, with help from [AUR PKGBUILD](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=megasync). I am new to nix so there may be some things that are not ideal.

Closes #308640 
## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

hackerncoder committed:
megasync: 4.9.1.0 -> 5.5.0.0

FliegendeWurst commented on pkgs/applications/misc/megasync/default.nix:
```diff
@@ -1,75 +1,91 @@
     cryptopp
     curl
     ffmpeg
+    freeimage
```
Is freeimage really required? It's marked as insecure.

hackerncoder replied:
No. I don't think so, it's "used to manage thumbnails/previews" (https://github.com/meganz/sdk/blob/master/examples/win32/README.md?plain=1#L50). I'll disable it.

hackerncoder committed:
megasync: remove freeimage

also use lib.cmakeBool

hackerncoder committed:
megasync: use fetchpatch

hackerncoder committed:
megasync: version fetchpatch urls so they don't magically break

hackerncoder committed:
megasync: 5.5.0.0 -> 5.6.1.0

As stated, the DesignTokensImporter submodule cannot be fetched, so I only check
out the megasdk submodule.

The get_clang_format fails, so copy it from local

hackerncoder commented:
As much as I hate this package, seeing as there is no maintainer and I will be using it in the near future, I am thinking about becoming maintainer.

FliegendeWurst commented on pkgs/applications/misc/megasync/default.nix:
```diff
@@ -1,75 +1,111 @@
     rev = "v${version}_Linux";
-    hash = "sha256-Y1nfY5iP64iSCYwzqxbjZAQNHyj4yVbSudSInm+yJzY=";
-    fetchSubmodules = true;
+    hash = "sha256-EjXmx+beYH3J60RidkEbZHFS5ZAOpJKmNC+WCIM84RA=";
```
```suggestion
    hash = "sha256-+022P3W6JXyEJS0vu++nl/SripVJ7LOjUAoCsu5yXhs=";
```
Are you sure your script below is reproducible?

hackerncoder committed:
megasync: don't cp clang-format

Spiker985 commented:
It appears that `legacyPackages.x86_64-linux.xorg.xrdb` is required in order to use Megasync 4.9.1+

Attempting to run `megasync --version` in a terminal results in:
```
Avoiding wayland
Error for "xrdb -query" command: "/run/current-system/sw/bin/bash: line 1: xrdb: command not found\n"
MEGAsync v4.9.1 (4aea5a1)

hackerncoder commented:
It is required for [megasync to figure out dpi](https://github.com/meganz/MEGAsync/blob/0ac3774efc2e6de5849d97e6dad6c552dc5b17f5/src/MEGASync/ScaleFactorManager.cpp#L15)

Spiker985 commented:
> It is required for [megasync to figure out dpi](https://github.com/meganz/MEGAsync/blob/0ac3774efc2e6de5849d97e6dad6c552dc5b17f5/src/MEGASync/ScaleFactorManager.cpp#L15)

Makes sense, is there any way to add it as a dependency as part of the package, since it's required for use? I have no concept as to if that's possible or not

FliegendeWurst commented:
Pretty easy. Just add to postInstall `wrapProgram $out/bin/megasync --prefix PATH : ${xorg.xrdb}/bin` (or similar, there are loads of examples in nixpkgs)

hackerncoder committed:
megasync: Wrap xorg.xrdb

hackerncoder commented:
I have tried to wrap xrdb, it compiles without errors, but I have xrdb in my path so I don't know if it actually works.

FliegendeWurst commented on pkgs/applications/misc/megasync/default.nix:
```diff
@@ -81,46 +116,24 @@ mkDerivation rec {
-      lrelease MEGASync.pro
-      DESKTOP_DESTDIR="$out" qmake PREFIX="$out" -o Makefile MEGASync.pro CONFIG+="nofreeimage release"
-    popd
+  postInstall = ''
+    wrapProgram $out/bin/megasync --prefix PATH : ${lib.makeBinPath [ xorg.xrdb ]}
```
```suggestion
  preFixup = ''
    qtWrapperArgs+=(--prefix PATH : ${lib.makeBinPath [ xorg.xrdb ]})
```
I had another look and apparently the Qt wrapper already wraps the program. To avoid double wrapping we should only add our custom arguments to the Qt wrapper.

FliegendeWurst commented:
To test, you run megasync in a `nix-shell --pure`.

FliegendeWurst committed:
megasync: avoid double wrapping
